### ISSUE TITLE
Should we prefix jQuery object variables with a $?

### ISSUE THREAD

**Issue by rafaellyra** [Werkspot/javascript#1]
Please, :+1: for yes or :-1:  for no.

update 1: (and also discuss if you want, of course)

**Comment by rafaellyra** [comment]
:+1:

**Comment by kasper-agg** [comment]
:-1:

**Comment by Terr** [comment]
With jQuery objects, do you mean  ```$buttonElements = $('.button');``` ?

**Comment by ruudvanderweijde** [comment]
doesn't matter to me, fine with both (well, with having one or the other)

**Comment by thijsBreker** [comment]
:-1: see @Terr's comment as to why

**Comment by kasper-agg** [comment]
Is the update SUM(tups - tdowns)? I just updated that way, but maybe you want to do +1 -3? :+1: for first approach... nah nevermind :+)

**Comment by rafaellyra** [comment]
not, it is just the first update on the initial issue :) I think we can sum it just looking :)

**Comment by kasper-agg** [comment]
Agree :)

**Comment by rafaellyra** [comment]
@Terr yes it is, I think that is easier to know where you can you all the jQuery methods and also helps when you receive it as an parameter:

//with $

```
function foo (event) {
    var $element = $(event.currentTarget);
    hide($element);
}


function hide ($element) {
    $element.hide();  // I know that I can do it
}
```

//without $

```
function foo (event) {
    var element = $(event.currentTarget);
    hide(element);
}

function hide (element) {
    element.hide(); // Can I do it? 
}
```

**Comment by kasper-agg** [comment]
Ah the serenity of the second part melts my heart!

**Comment by Terr** [comment]
@rafaellyra I see your point, and in JS you are missing the luxury of type hinting (i.e. specifying "jQueryObject element" in your function signature).

I think that the variable name itself should tell you about what's in it. If the name is consistently used then word 'element' in the variable name should tell you what kind of object it is. With a clear naming convention, '$' would be redundant.

**Comment by caseycs** [comment]
:+1: 
More strict rules = easier to read

**Comment by MarijnKoesen** [comment]
:+1:

Im used to prefix them with $ as well, as its much more clear whether you have a dom element or a jquery object.

**Comment by MarijnKoesen** [comment]
Re @ @terr comment: "But I think that the variable name itself should tell you about what's in it"

Thats exactly the reason Rafael opened the issue. To have a definition on how to name them. 

"With a clear naming convention, the '$' would be redundant."

But the thing is that the $ IS a naming conventionele making the 'element' postfix redundant. 

The $ is shorter, and at the front of the name, so you immediately see that its a jquery object.

Also, with the $ you van specifically say its a jquery object, if you do it with postfixing 'element' you cannot mark something as jquery.

So if you have a list of buttons, what do you call the variable? `buttonElements` because you postfix them with element? I think `buttons` is better, shorter, clearer. Than jquery list will be `$buttons`.

Also if we decide the ditch jquery we don't need to rename all `elements` variables back to e.g. `buttons` because its isolated only in all the jquery specific code, because we only prefix jqeury code. Makes it easier to refactor and get rid of jquery for example, just search for all variables starting with $.

Ideally we have proper typehinting and don't need this, but since we don't I think $ is the better naming convention.

**Comment by snoek09** [comment]
Pick one and stick to it.

**Comment by MarijnKoesen** [comment]
If nobody has better arguments on why not to use it than my arguments to use it, we should/will use it.
